src/mpl/src/thread/mpl_thread_posix.c

```diff
@@ -82,4 +83,48 @@ void *MPLI_thread_start(void *arg)
     return NULL;
 }
 
+void MPL_thread_set_affinity(MPL_thread_id_t thread, int *affinity_arr, int affinity_size,
+                             int *errp)
+{
+#ifdef MPL_HAVE_PTHREAD_SETAFFINITY_NP
+    int err = MPL_SUCCESS, pthread_err;
+    int proc_idx, set_size = 0;
+    cpu_set_t cpuset;
+    __CPU_ZERO_S(sizeof(cpu_set_t), &cpuset);
+
+    for (proc_idx = 0; proc_idx < affinity_size; proc_idx++)
+        __CPU_SET_S(affinity_arr[proc_idx], sizeof(cpu_set_t), &cpuset);
+
+    if ((pthread_err = pthread_setaffinity_np(thread, sizeof(cpu_set_t), &cpuset)) != 0) {
+        MPL_error_printf("pthread_setaffinity_np failed\n");
+        err = MPL_ERR_THREAD;
+        goto fn_exit;
+    }
+
+    if ((pthread_err = pthread_getaffinity_np(thread, sizeof(cpu_set_t), &cpuset)) != 0) {
+        MPL_error_printf("pthread_getaffinity_np failed\n");
+        err = MPL_ERR_THREAD;
+        goto fn_exit;
+    }
+
+    for (proc_idx = 0; proc_idx < affinity_size; proc_idx++) {
+        if (__CPU_ISSET_S(affinity_arr[proc_idx], sizeof(cpu_set_t), &cpuset))
+            set_size++;
+    }
+    if (set_size != affinity_size) {
+        MPL_error_printf("affinity is not set\n");
```

Comment:
Remove all the `MPL_error_printf`. The caller will handle the error, right?

Reply:
Missed these. Will remove them.

Reply:
Done